lib/pluginmanager/unpack.rb

```diff
@@ -0,0 +1,47 @@
+class LogStash::PluginManager::Unpack < LogStash::PluginManager::Command
+
+  option "--tgz", :flag, "unpack a packaged tar.gz file", :default => !LogStash::Environment.windows?
+  option "--zip", :flag, "unpack a packaged  zip file", :default => LogStash::Environment.windows?
+  option "--force", :flag, "Override any previously generated cache", :default => false
+
+  parameter "file", "the package file name", :attribute_name => :package_file, :required => true
+
+  def execute
+    puts("Unpacking #{package_file} to be used as a plugins local cache")
+    if force?
+      require "fileutils"
+      FileUtils.rm_rf(LogStash::Environment::CACHE_PATH)
+    end
+    validate_cache_location
+    unpack(package_file)
+    puts("Unpacked at #{LogStash::Environment::CACHE_PATH}")
+    puts("If you want to use the generated cache to install/update plugins you should call bin/plugin with the --local flag")
+  end
+
+  private
+
+  def validate_cache_location
+    require "fileutils"
+    location = LogStash::Environment::CACHE_PATH
+    if File.exist?(location)
+      puts("Directory #{location} is going to be override, do you want to continue? (Y/N)")
+      override = ( "y" == STDIN.gets.strip.downcase ? true : false)
+      if override
+        FileUtils.rm_rf(location)
+      else
+        puts("Unpack cancelled, a previously generated cache exist at location: #{location}, move it to a save place and run the command again")
+        exit
```

Comment:
Is not an error the user decided to cancel, not logstash.

PD: This PR stars to look like the https://github.com/elastic/logstash/pull/3111 I think we will write here even more comments :+1: